nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+/* *****************************************************************************
+ * Copyright (c) 2015-2018 Konduit k.k.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.linalg.factory.ops;
+
+import org.junit.Test;
+import org.nd4j.linalg.BaseNd4jTest;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.factory.Nd4jBackend;
+import org.nd4j.linalg.indexing.conditions.Conditions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class NDBaseTest extends BaseNd4jTest {
+    public NDBaseTest(Nd4jBackend backend) {
+        super(backend);
+    }
+
+    @Test
+    public void testAll() {
+        NDBase base = new NDBase();
+
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.all(x, 1);
+        INDArray y_exp = Nd4j.create(new boolean[]{false, false, false});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
```

Comment:
`assertEquals(y_exp, y)`
Don't need equalWithEps for boolean and integer types; furthermore assertEquals aids debugging (it prints both values if test fails, whereas assertTrue can't/doesn't)
I won't flag other instances, but there's a few (testAny, testArgmax, etc)
also FYI INDArray.equals requires the datatypes to match, so you don't need to separately do assertEquals(<datatypes>)